### PR TITLE
Fix running 'wheelhouse' test script on Travis CI

### PR DESCRIPTION
**Pull request by takluyver**
This is an allowed failure, but so long as it's there, it might as well work.

### CODE REVIEW

**Comment by takluyver** [comment]
@njsmith: this change would have the CI hitting your vorpus.org server - the code was there before, but commented out. I'd like your approval before merging it.

@rmcgibbo: your 'wheelhouse' at stanford.edu is no longer available. Let me know if you have the same files at some other URL. I don't think this is high priority, so don't spend energy on it, but if it's just a case of tweaking the URL, we can do that.

**Comment by rmcgibbo** [comment]
I don't have those files anymore (Stanford took down my webspace a couple months after I graduated), but they weren't anything special.

**Comment by rmcgibbo** [comment]
@takluyver: from these [these](https://travis-ci.org/pypa/auditwheel/builds/176840549) travis logs from 3 years ago, it looks to me like the tests were covering wheels from `lxml-3.7.3`, `Pillow-4.1.1`, `PyYAML-3.12`.

**Comment by takluyver** [comment]
Thanks! Do you think it's worth re-creating some more tests with real-world wheels, or is it more a nice-to-have alongside the main test suite? I wanted to make sure that whatever's there is running effectively, but I don't know how important it is.

File: tests/integration/test-wheelhouses.sh [hunk 1]
```diff
@@ -1,19 +1,8 @@
...
-done
-
 # Download some other wheels that Nathaniel made
-#URL=https://vorpus.org/~njs/tmp/manylinux-test-wheels/original/
-#curl --silent $URL | grep 'a href' | grep whl | cut -d ' ' -f 8 | cut -d '=' -f 2 | cut -d '"' -f 2 | xargs -n1 -I '{}' wget --no-check-certificate -P wheelhouse-njs "$URL/{}"
-
-# Download some more wheels that Robert made
-URL=http://stanford.edu/~rmcgibbo/wheelhouse/
-curl --silent $URL | grep 'a href' | grep whl | cut -d ' ' -f 8 | cut -d '=' -f 2 | cut -d '"' -f 2 | xargs -n1 -I '{}' wget -P wheelhouse-rmcgibbo "$URL/{}" -nc
+URL=https://vorpus.org/~njs/tmp/manylinux-test-wheels/original/
```
Comment (lkollar) [inline tests/integration/test-wheelhouses.sh]: Why not include these wheels in the source repo instead? We already have a few under the tests directory.

**Comment by lkollar** [comment]
This should fix #142. We should also remove the `allow_failures` bit in `.travis.yml`.

File: tests/integration/test-wheelhouses.sh (hunk 1 above)
Comment (takluyver) [reply tests/integration/test-wheelhouses.sh]: I wouldn't want to clutter up the source repo with various multi-MB wheels; large binary files are a bit awkward to handle in git. This test can also easily be expanded to many more wheels, e.g. I saw [this scikit-learn wheelhouse](https://3f23b170c54c2533c070-1c8a9b3114517dc5fe17b7c3f8c63a43.ssl.cf2.rackcdn.com/) linked from #86.

**Comment by takluyver** [comment]
The latest commit removes the `allow_failures` section.

I'm in two minds about this. On the one hand, it could be a flaky test, as it relies on downloading extra files. But then I don't think `allow_failures` tests are much use. At least the way I use Travis, I usually ignore them, and as these have been failing for about 18 months, it seems I'm not the only one.

In any case, I'd still like an OK from @njsmith before we merge this, because it will hit his server.

File: tests/integration/test-wheelhouses.sh (hunk 1 above)
Comment (mayeut) [reply tests/integration/test-wheelhouses.sh]: One way not to clutter up the source repo with various multi-MB wheels would be to use `git-lfs`.
> Do you think it's worth re-creating some more tests with real-world wheels, or is it more a nice-to-have alongside the main test suite? I wanted to make sure that whatever's there is running effectively, but I don't know how important it is.

Well, I think that's the main question here. I don't know if it's important either... Were those tests added to cover a feature, bug, in the past, I can't say looking at the history...

**Comment by mayeut** [comment]
> Do you think it's worth re-creating some more tests with real-world wheels, or is it more a nice-to-have alongside the main test suite? I wanted to make sure that whatever's there is running effectively, but I don't know how important it is.

Well, I think that's the main question here. I don't know if it's important either... Were those tests added to cover a feature, bug, in the past, I can't say looking at the history...

**Comment by rmcgibbo** [comment]
I added those tests, but it was 3 years ago so I don't remember my exact train of thought. Generally though, at that time this project was something I was just hacking on. I don't think it was an official pypa project, I don't think that the first manylinux pep had yet been merged (it was probably under discussion), and it wasn't clear whether this software would even fill the intended use case. There wasn't really a "main test suite". I think I basically figured "if I can process some random wheels that I found on the internet, then it's probably working, and that's easier than investing in writing a bunch of unit tests right now"

So... real-world tests are always great. But I'd recommend making the decision on whether or not to include tests like this based on your current assessment of the coverage of the test suite, not some no-longer-applicable-determination I made three years ago when this was a solo hobby project that I was hacking on while procrastinating on my dissertation.

**Comment by takluyver** [comment]
Thanks @rmcgibbo. That's the sort of scenario I would have guessed that script came from. :-)

@lkollar is doing some good work towards increasing test coverage. But I don't yet have a good feel for how well the test suite exercises gnarly cases that might come up in real-world wheels.

**Comment by lkollar** [comment]
Given what @rmcgibbo said it might not be useful to try to fix these tests with the same wheels. The `test-wheelhouse.sh` script does not try to import the repaired wheels either, so this test is not that useful anyway. The repair step's return value is not even checked so we won't catch issues there. We now have several more exhaustive tests which validate specific features and actually test that the resulting libraries are usable. So I think we should consider two alternatives here:

- Remove these tests altogether and keep adding integration tests for wheels for the gnarly cases.  https://github.com/pypa/auditwheel/issues/159  is an example but we would have to xfail it until `patchelf` lands the fix. I'm sure we could find more in the issue tracker.
- Keep the wheels but replace the shell script with a proper Python test which repairs and imports the modules. 

If we keep the wheel I still think that we should store them in the source repo. [pip](https://github.com/pypa/pip/tree/master/tests/data/packages) does this too and it seems to be working just fine.

**Comment by takluyver** [comment]
Now we've got coverage measurement set up and we can see we're exercising most of the code already, I think I'm comfortable with removing the wheelhouses script, and #179 does that.

**Comment by takluyver** [comment]
#179 has been merged instead of this.

**Pull request closed without merging**